infra/aws/terraform/cncf-k8s-infra-aws-capa-ami/iam-capa-maintainers-group.tf

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_iam_group" "maintainers" {
+  name = "maintainers"
+}
+
+resource "aws_iam_group_membership" "maintainers" {
+  name = "capa-maintainers"
+
+  users = [
+    aws_iam_user.ankitasw.name,
+    aws_iam_user.dlipovetsky.name,
+    aws_iam_user.richardcase.name,
+    aws_iam_user.vincepri.name,
+  ]
+
+  group = aws_iam_group.maintainers.name
+}
+
+resource "aws_iam_group_policy_attachment" "maintainer-admin" {
+  group      = aws_iam_group.maintainers.name
+  policy_arn = "arn:aws:iam::aws:policy/AdministratorAccess"
```

Comment:
Is there a least-privilege alternative to this access that we can grant?

Reply:
Yes for sure. How about initially we scope the "admin" down to only the services that are used for AMI publishing. The original thinking was that more elevated permissions would be needed initially whilst we work out all the automation / diagnose issues.

Ultimately, we could make it so it's just the build agent (prow or GHA) that has access.

Reply:
TBH, I don't even agree with giving `AdministratorAccess` to Prow.

Reply:
The more i think about it (based on your feedback), the maintainers don't need access to the account. Just whatever CI we use. I'll also re-think this based on that.